### PR TITLE
add unit tests for json-utilities and cover some exceptions #57

### PR DESCRIPTION
**Pull request by anilbey**
- adds some missing unit tests

-  separates the test_protocols into two files
    * one for protocols.reader
    * the other for protocols.create_protocols

### CODE REVIEW

File: tests/unit_tests/test_json_utilities.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    assert test_np_array_decoded == test_np_array.tolist()
+
+
+def test_np_encoder_integer():
+    """Unit test for the NpEncoder on np.integer."""
+    test_np_integer = np.int16(1)
+    test_np_integer_encoded = json.dumps(test_np_integer, cls=NpEncoder)
+    test_np_integer_decoded = json.loads(test_np_integer_encoded)
+    assert test_np_integer_decoded == test_np_integer
+
```
Comment (AurelienJaquier) [inline tests/unit_tests/test_json_utilities.py:22]: Very nice! Could also add a similar test function for numpy floats, like np.float32, for coverage.

Comment (anilbey) [reply tests/unit_tests/test_json_utilities.py:22]: Thanks :)

**Comment by anilbey** [comment]
I was hoping this would bring the coverage above %70

File: tests/unit_tests/test_json_utilities.py (hunk 1 above)
Comment (anilbey) [reply tests/unit_tests/test_json_utilities.py:22]: added in the next commit

**Review by AurelienJaquier — APPROVED** [review]

**Pull request merged**